redbot/core/commands/commands.py

```diff
@@ -809,6 +820,100 @@ def help(self):
         if doc:
             return inspect.cleandoc(translator(doc))
 
+    async def red_get_data_for_user(self, *, user_id: int) -> MutableMapping[str, io.BytesIO]:
+        """
+        This should be overridden by cogs
+        that support users requesting their data
+        as well as cogs that don't store data.
+
+        Overriden implementations should send a io.BytesIO containing a
+        human-readable version of the data
+        the cog has about the specified user_id or commands.RED_NO_EUD
+        if the cog does not have end user data.
+
+        The data should be easily understood for what it represents to
+        most users of age to use Discord.
+        You may include multiple files by providing an archive,
+        including a readme file which explains anything additional needed.
+
+        This method may also be implemented for an extension.
+
+        Parameters
+        ----------
+        user_id: int
+
+        Returns
+        -------
+        MutableMapping[str, io.BytesIO]
+            A mapping of filenames to BytesIO objects
+            suitable to send as a files or as part of an archive to a user.
+
+            .. note::
+                Cogs which explicitly do not store EUD should return
```

Comment:
Is this return *just* returning `commands.RED_NO_UED` (NOT in a MutableMapping)? Maybe add this to the return hints? `typing.Union[MutableMapping[str, io.BytesIO], commands.RED_NO_UED]`

Reply:
Can't put that in the type hint, type would be

`typing.Union[MutableMapping[str, io.BytesIO], object]` at that point.

Reply:
So, I've updated this a bit more. `commands.RED_NO_EUD` is gone. If `typing.Literal` where a bit more powerful, I'd be happier about special sentinel values.